### DISCUSSION TITLE
Insert data from a uploaded CSV file

### DISCUSSION THREAD

**Discussion by Vipul-Cariappa** [sqlpage/SQLPage#493] [Q&A] [answered]
I want to insert data into DB using an uploaded CSV file. How do I do it?

My table structure:
```sql
CREATE TABLE Student(
    ID INTEGER PRIMARY KEY AUTOINCREMENT,
    RegNo CHAR(12) UNIQUE NOT NULL,
    Name VARCHAR(50) NOT NULL,
    Section VARCHAR(10) NOT NULL,
    GNo INTEGER,
    Interim INTEGER DEFAULT 0,
    Final INTEGER DEFAULT 0,
    Report INTEGER DEFAULT 0,
    CONSTRAINT fk
        FOREIGN KEY (Section) REFERENCES Section(Name),
        FOREIGN KEY (GNo) REFERENCES Team(ID)
        ON DELETE CASCADE
);
```

The user would only update the `RegNo`, `Name`, and `Section` columns. Other columns will be set using defaults or to `NULL`.

I am able to get the contents of CSV file by doing this
```sql
SET $csv_data = sqlpage.read_file_as_text(sqlpage.uploaded_file_path('csvFile'));

SELECT
    'text' AS component,
    $csv_data AS contents
WHERE
    $csv_data IS NOT NULL;

SELECT
    'form' AS component;
```

But have no idea how to proceed.

> I will be using sqlite.

---

SQLPages is an amazing project. Thanks.

**Comment by lovasoa** [comment, marked as answer]
Hello and welcome to SQLPage !

Did you take a look at *Example 11: Bulk data insertion* in https://sql.ophir.dev/documentation.sql?component=form#component ?

**Reply by Vipul-Cariappa** [reply]
Thanks a lot that example is exactly what I wanted.
Sorry, I did not go through the documentation properly.
